persistence-api/src/main/java/io/stargate/db/datastore/DataStore.java

```diff
@@ -151,6 +151,11 @@ default QueryBuilder queryBuilder() {
    */
   Schema schema();
 
+  /** Returns true if the persistence backend supports secondary indexes */
+  default boolean supportsSecondaryIndex() {
+    return true;
```

Comment:
Having two defaults for the same flag seems a bit error-prone to me :thinking: how about we keep the default on `Persistence` and `DataStore` impl. will redirect there?

Reply:
I was kinda trying to do that - but since according to this hierarchy it's possible that a DataStore does not have a Persistence (i.e. according to this type structure not all DataStores are PersistenceBackedDataStores), I needed a value for `supportsSecondaryIndex` when the DataStore is theoretically not backed by a persistence.

But is there a real situation where a DataStore isn't backed by Persistence? That kind of confused me

Reply:
In tests I did have `DataStore` without a `Persistence`, but in real implementations is it conceptually associated with `Persistence` I think.